src/services/auth.service.js

```diff
@@ -0,0 +1,52 @@
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
+
+const userDao = require("../models/user.dao");
+
+const signUp = async (
+  name,
+  birthdate,
+  phoneNumber,
+  gender,
+  address,
+  email,
+  password
+) => {
+  const user = await userDao.getUserByEmail(email);
+
+  if (user !== undefined) {
+    throw new Error("EMAIL_ALREADY_TAKEN");
+  }
```

Comment:
다음과 같이 조건문 내부의 조건을 더욱 줄여서 사용할 수 있습니다.
그리고 회원가입이니 email이 있다는 메시지를 반환하기 보단 해당 이메일을 가진 유저가 있다는 것을 보여주는 것이 좋아보입니다.

만약 이메일 중복을 방지하기 위해서는 `user.email`이 input으로 받은 email 데이터와 일치하는지 확인할 필요가 있을 것 같습니다!

```suggestion
  if (user) {
    throw new Error("USER_ALREADY_EXIST");
  }
```